packages/ui-components/src/components/NativeSelect/NativeSelect.component.tsx

```diff
@@ -0,0 +1,254 @@
+  /**
+   * Additional CSS classes to apply to the select element for custom styling.
+   */
+  className?: string
+
+  /**
+   * Elements to be rendered inside the select element.
+   * This can be any React node or a collection of React nodes.
+   * Typically, these are SelectOption or SelectOptionGroup components.
+   */
+  children?: ReactNode
+
+  /**
+   * Disables the select element, making it unclickable.
+   */
+  disabled?: boolean
+
+  /**
+   * Highlights the select element as invalid, indicating incorrect user input or validation errors.
+   */
+  invalid?: boolean
+
+  /**
+   * Highlights the select element as valid, indicating correct user input or successful validation.
+   */
+  valid?: boolean
+
+  /**
+   * Displays a loading indicator over the select element.
+   * Used for async operations like fetching data.
+   */
+  loading?: boolean
+
+  /**
+   * Displays an error state over the select element, such as during data fetching errors.
+   * Should not be used for validation errors.
+   */
+  error?: boolean
+
+  /**
+   * Event handler for the change event of the select element.
+   * Triggered when the user changes the selected option.
+   */
+  // eslint-disable-next-line no-unused-vars
+  onChange?: (event: ChangeEvent<HTMLSelectElement>) => void
+
+  /**
+   * Event handler for the click event on the select element.
+   * Triggered when the user clicks on the select element.
+   */
+  // eslint-disable-next-line no-unused-vars
+  onClick?: (event: MouseEvent<HTMLSelectElement>) => void
+
+  /**
+   * Additionl CSS classes to apply to the outer wrapper of the select component for custom styling.
+   */
+  wrapperClassName?: string
+}
+
+/**
+ * Renders the appropriate 'state' icon based on the state of the select element
+ */
+const SelectIcons: FC<{
+  isLoading: boolean
+  hasError: boolean
+  isInvalid: boolean
+  isValid: boolean
+  disabled?: boolean
+}> = ({ isLoading, hasError, isInvalid, isValid, disabled }) => {
+  if (isLoading) {
+    return (
+      <div className={`juno-select-loading ${stateClasses.loading}`}>
+        <Spinner className={stateClasses.loadingSpinner} />
+      </div>
+    )
+  }
+  if (hasError) {
+    return (
+      <div className={`juno-select-error-text ${stateClasses.error}`}>
+        <Icon icon="errorOutline" color="jn-text-theme-error" className={stateClasses.errorIcon} />
+      </div>
+    )
+  }
+  return (
+    <div className={`${iconBaseClasses} ${disabled ? iconDisabledClasses : ""}`}>
+      {isInvalid && <Icon icon="dangerous" color="jn-text-theme-error" />}
+      {isValid && <Icon icon="checkCircle" color="jn-text-theme-success" />}
+      <Icon icon="expandMore" />
+    </div>
+  )
+}
+
+/**
+ * A basic, uncontrolled native HTML select component.
+ * Takes NativeSelectOption and NativeSelectOptionGroup children as options.
+ * Handles states such as loading, valid, invalid, and error.
+ */
+export const NativeSelect: FC<NativeSelectProps> = ({
+  name = "Unnamed Select",
```

Comment:
In the original version, the value was set to null. Is there a specific reason for changing it to a default String value?

Reply:
We didn't assign it as a default value, [but in our code](https://github.com/cloudoperators/juno/blob/eea5de67a22b0429506040285cde1d1e2d35739d/packages/ui-components/src/components/NativeSelect/NativeSelect.component.js#L194), it's technically the default value. Seemed correct to let the user know explicitly. What do you think?

Reply:
The name is sent as a property in forms. While it’s unlikely that we’ll use forms in the React context, if we do, the name will be set, which could potentially confuse developers. If this value is null, the selected option will not be included when the form is submitted. This is the expected behavior, and I wouldn’t recommend changing it.

![image](https://github.com/user-attachments/assets/c2940de9-4f85-4232-8ae3-328243d9a058)

Reply:
Will definitely change it. 🙈 Just really wanna understand what I'm missing first.
In old code, we initially set `name` as `null` by default. [But then in our code](https://github.com/cloudoperators/juno/blob/eea5de67a22b0429506040285cde1d1e2d35739d/packages/ui-components/src/components/NativeSelect/NativeSelect.component.js#L194), if `name` isn't set, we always set it to  `Unnamed Select` (as a **hidden default**).

I just wanted to make the default explicit so it indeed doesn't confuse the user.
But maybe I'm missing something in between, how setting the default initially changes the overall behaviour? 🙈

**Side note:**
Now that you bring it up though... Maybe we should remove the default `name` value from code all together? To make sure the option is only included in form submissions if explicitly set? Could add to future improvements.

Reply:
Let's do improvements after migration.

Reply:
Added https://github.com/cloudoperators/juno/issues/240#issuecomment-2468482674